kubernetes/upgrade/checks.go

```diff
@@ -118,11 +123,52 @@ func NewChecks(path *Path, state state.State, k8sConfig *rest.Config, controlPla
 					"DynamicKubeletConfig",
 				},
 			},
+			// https://kubernetes.io/blog/2023/03/17/upcoming-changes-in-kubernetes-v1-27/
+			"1.26->1.27": {
+				kubeAPIServerChecks: apiServerCheck{
+					removedAPIResources: []string{
+						"csistoragecapacities.v1beta1.storage.k8s.io",
+					},
+					componentCheck: componentCheck{
+						removedFlags: []string{
+							"master-service-namespace",
```

Comment:
I checked this one, and it seems this flag was actually removed from kubelet, not kube-apiserver and [this page](https://kubernetes.io/blog/2023/03/17/upcoming-changes-in-kubernetes-v1-27/#removal-of-master-service-namespace-command-line-argument) has wrong information, unless if I am missing something.

Reply:
both are removed, apiserver and kubelet https://github.com/kubernetes/kubernetes/pull/116015 see this PR

Reply:
the page actually missed information on removed kubelet flags :sweat_smile:

Reply:
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#other-cleanup-or-flake it's in the changelog though :sweat_smile:

Reply:
Ah, so it seems it was removed from `kube-apiserver` already on `1.25->1.26`, and now it is also removed from `kubelet` on `1.26->1.27`.

Reply:
oh it was removed way back, in k8s 1.7, i missed that, good catch, I'll fixup the PR, yes, the link is wrong it seems

Reply:
cool. just a small note: I downloaded the binaries of apiserver of various versions and checked myself - the flag seems to be removed from it exactly on `1.25->1.26`, not way back in 1.7.

Reply:
sure in 1.26? https://github.com/kubernetes/kubernetes/pull/114446 this says 1.27 thinking

Reply:
🤔 

```
$ $ ./kube-apiserver-1.25 --version
Kubernetes v1.25.0
$ ./kube-apiserver-1.25 --help | grep master-service
      --master-service-namespace string                DEPRECATED: the namespace from which the Kubernetes master services should be injected into pods. (default "default")
$ ./kube-apiserver-1.26 --version
Kubernetes v1.26.3
$ ./kube-apiserver-1.26 --help | grep master-service
$ ./kube-apiserver-1.27 --version
Kubernetes v1.27.0-beta.0
$ ./kube-apiserver-1.27 --help | grep master-service
$
```

Reply:
yeh, seems it got removed in 1.26, but the PR mentions 1.26, seems a little confusion